packages/keyring-controller/src/KeyringController.test.ts

```diff
@@ -586,17 +612,33 @@ describe('KeyringController', () => {
               },
             );
           });
-        });
 
-        cacheEncryptionKey &&
-          it('should set encryptionKey and encryptionSalt in state', async () => {
-            // TODO: Either fix this lint violation or explain why it's necessary to ignore.
-            // eslint-disable-next-line @typescript-eslint/no-floating-promises
-            withController({ cacheEncryptionKey }, async ({ initialState }) => {
-              expect(initialState.encryptionKey).toBeDefined();
-              expect(initialState.encryptionSalt).toBeDefined();
+          !cacheEncryptionKey &&
+            it('should remove encryptionKey and encryptionSalt from state when cacheEncryptionKey is false', async () => {
```

Comment:
Really we're not testing that they were removed, because in this scenario it's impossible for them to be set in the first place.


```suggestion
            it('should not set encryptionKey and encryptionSalt in state', async () => {
```

Reply:
The same applies to the version of this test above as well